homeassistant/components/hunterdouglas_powerview/cover.py

```diff
@@ -56,14 +56,38 @@ async def async_setup_entry(
 
     pv_entry: PowerviewEntryData = hass.data[DOMAIN][entry.entry_id]
     coordinator: PowerviewShadeUpdateCoordinator = pv_entry.coordinator
+    cancel_resync: CALLBACK_TYPE | None = None
+
+    async def _async_refresh_after_import(self, *_: Any) -> None:
+        """Force position refresh shortly after adding.
+
+        Legacy shades can become out of sync with hub when moved
+        using physical remotes. This also allows reducing speed
+        of calls to older generation hubs in an effort to
+        prevent hub crashes.
+        """
+
+        for shade in pv_entry.shade_data.values():
+            with suppress(asyncio.TimeoutError):
+                # hold off to avoid spamming the hub
+                async with asyncio.timeout(10):
+                    _LOGGER.debug("Initial refresh of shade: %s", shade.name)
+                    await shade.refresh()
```

Comment:
It would be better if the timeout was handled by aiohttp instead of wrapping with asyncio.timeout here since there are technically two time outs running

Each request can have a different timeout https://docs.aiohttp.org/en/stable/client_reference.html#aiohttp.ClientSession.request

Reply:
sorry mate - i am really time poor and doing this one outside of work. appreciate if you can point me to what you are after here
Are you suggesting we handle the timeout in the upstream library by reworking the below ?
```
async def refresh(self):
        """Query the hub and the actual shade to get the most recent shade
        data. Including current shade position."""
        try:
            _LOGGER.debug("Refreshing position of: %s", self.name)
            raw_data = await self.request.get(self._resource_path, {"refresh": "true"})
            # Gen <= 2 API has raw data under shade key.  Gen >= 3 API this is flattened.
            self._raw_data = raw_data.get(ATTR_SHADE, raw_data)
        except PvApiMaintenance:
            _LOGGER.debug("Hub undergoing maintenance. Please try again")
        return
```

Reply:
> sorry mate - i am really time poor and doing this one outside of work. 
Seems like we all have that problem 😉
> appreciate if you can point me to what you are after here Are you suggesting we handle the timeout in the upstream library by reworking the below ?
> 
> I would think we want to keep those timeouts visible (for cases where there is an issue)
> 
> ```
> async def refresh(self):
>         """Query the hub and the actual shade to get the most recent shade
>         data. Including current shade position."""
>         try:
>             _LOGGER.debug("Refreshing position of: %s", self.name)
>             raw_data = await self.request.get(self._resource_path, {"refresh": "true"})
>             # Gen <= 2 API has raw data under shade key.  Gen >= 3 API this is flattened.
>             self._raw_data = raw_data.get(ATTR_SHADE, raw_data)
>         except PvApiMaintenance:
>             _LOGGER.debug("Hub undergoing maintenance. Please try again")
>         return
> ```

I think the simplest way is to pass timeout to the refresh call in the lib and than pass that timeout down to the aiohttp get request

Reply:
Pretty much all the asyncio.timeout wrapping are comprises before we had access to the lib. Using aiohttp to do the timeout is much safer because when we use an external timeout we have to hope every bit of code running inside the timeout is cancellation safe which is frequently not the case and can lead to seemingly unexplained problems

Reply:
I'm going to merge this as-is. The `asyncio.timeout`s can be adjusted in a future PR. We should remove all of them an let `aiohttp` do the time outs in a separate PR anyways.